File: java/app/src/main/java/com/youknow/movie/ui/common/GridItemDecoration.java
package com.youknow.movie.ui.common;

import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class GridItemDecoration extends RecyclerView.ItemDecoration {

    private final int padding;

    public GridItemDecoration(int padding) {
        this.padding = padding / 2;
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        if (parent.getPaddingLeft() != padding) {
            parent.setPadding(padding, padding, padding, padding);
            parent.setClipToPadding(false);
        }

        outRect.top = padding;
        outRect.bottom = padding;
        outRect.left = padding;
        outRect.right = padding;
    }
}

File: kotlin-databinding-coroutine-aac/app/src/main/java/com/youknow/movie/ui/common/GridItemDecoration.kt
package com.youknow.movie.ui.common

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class GridItemDecoration(private var padding: Int) : RecyclerView.ItemDecoration() {

    init {
        padding /= 2
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        if (parent.paddingLeft != padding) {
            parent.setPadding(padding, padding, padding, padding)
            parent.clipToPadding = false
        }

        outRect.apply {
            top = padding
            bottom = padding
            left = padding
            right = padding
        }
    }
}